Lib/asyncio/tasks.py

```diff
@@ -53,7 +53,10 @@ def current_task(cls, loop=None):
         """
         if loop is None:
             loop = events.get_event_loop()
-        return cls._current_tasks.get(loop)
+        try:
+            return loop.current_task()
+        except (AttributeError, NotImplementedError):
+            return cls._current_tasks.get(loop)
 
     @classmethod
     def all_tasks(cls, loop=None):
```

Comment:
Let's raise `PendingDeprecationWarning` in `Task.all_tasks()` and `Task.current_task()`, suggesting to use the new event loop methods and that `Task` methods will be removed in 3.9.

Reply:
Done.